cmake/FindCython.cmake

```diff
@@ -28,7 +28,7 @@ find_package(Python 3.6)
 if(Python_Interpreter_FOUND)
   get_filename_component( _python_path ${Python_EXECUTABLE} PATH )
   find_program( CYTHON_EXECUTABLE
-      NAMES cython${Python_VERSION_MAJOR} cython-${Python_VERSION_MAJOR}.${Python_VERSION_MINOR} cython cython.bat
+	  NAMES cython cython.bat cython${Python_VERSION_MAJOR} cython-${Python_VERSION_MAJOR}.${Python_VERSION_MINOR} cython3
```

Comment:
I wouldn't change the order of these, cython3 should always be preferred to plain cython (which might be python2)

Reply:
Thanks @rtburns-jpl. It looks like the problem is that on the machine I'm trying (dgx) there is a pre-installed ```cython3``` at ```/usr/local/bin/cython3```. The cython in conda bin directory is just called ```cython``` not ```cython3```. So I guess that is the issue why the one from system is chosen if I don't change the order. Any suggestion to force the cython from conda even though it is called cython and not cython3?

Reply:
I think you can explicitly override with e.g. `-DCYTHON_EXECUTABLE=$CONDA_PREFIX/bin/cython`

We could also try `${Python_EXECUTABLE} -c cython` like we do in isce3, which should more reliably pick up the right cython installation

Reply:
```-DCYTHON_EXECUTABLE=$CONDA_PREFIX/bin/cython``` worked fine. So I reverted the order as you suggested.